runtime/jcl/common/mgmtthread.c

```diff
@@ -1287,8 +1292,28 @@ freeThreadInfos(J9VMThread *currentThread, ThreadInfo *allinfo, UDATA allinfolen
 		while (sinfo) {
 			next = sinfo->next;
 			j9mem_free_memory(sinfo);
+			if (NULL != sinfo->obj.unsafe) {
+				vmfns->j9jni_deleteLocalRef((JNIEnv *)currentThread, (jobject) sinfo->obj.unsafe);
+			}
 			sinfo = next;
 		}
+		if (NULL != allinfo[i].thread) {
```

Comment:
The delete code handles a NULL reference as input, so the NULL checks can be removed.

Reply:
done.